src/commissaire_service/clusterexec/__init__.py

```diff
@@ -0,0 +1,305 @@
+                'model_json_data': model_instance.to_json()
+            }
+            self.request('storage.save', params=params)
+        except Exception as error:
+            self.logger.error(
+                'Unable to save final state for "{}" clusterexec '
+                'due to {}: {}'.format(cluster_name, type(error), error))
+
+    def on_upgrade(self, message, cluster_name):
+        """
+        Executes an upgrade command on hosts across a cluster.
+
+        :param message: A message instance
+        :type message: kombu.message.Message
+        :param cluster_name: The name of a cluster
+        :type cluster_name: str
+        """
+        command_args = ('upgrade',)
+        model_instance = ClusterUpgrade.new(
+            name=cluster_name,
+            status='in_process',
+            started_at=datetime.datetime.utcnow().isoformat(),
+            upgraded=[],
+            in_process=[]
+        )
+        self._execute(model_instance, command_args, 'upgraded')
+
+    def on_restart(self, message, cluster_name):
+        """
+        Executes a restart command on hosts across a cluster.
+
+        :param message: A message instance
+        :type message: kombu.message.Message
+        :param cluster_name: The name of a cluster
+        :type cluster_name: str
+        """
+        command_args = ('restart',)
+        model_instance = ClusterRestart.new(
+            name=cluster_name,
+            status='in_process',
+            started_at=datetime.datetime.utcnow().isoformat(),
+            restarted=[],
+            in_process=[]
+        )
+        self._execute(model_instance, command_args, 'restarted')
+
+    def on_deploy(self, message, cluster_name, version):
+        """
+        Executes a deploy command on atomic hosts across a cluster.
+
+        :param message: A message instance
+        :type message: kombu.message.Message
+        :param cluster_name: The name of a cluster
+        :type cluster_name: str
+        :param version: The tree image version to deploy
+        :type version: str
+        """
+        command_args = ('deploy', version)
+        model_instance = ClusterDeploy.new(
+            name=cluster_name,
+            status='in_process',
+            started_at=datetime.datetime.utcnow().isoformat(),
+            version=version,
+            deployed=[],
+            in_process=[]
+        )
+        self._execute(model_instance, command_args, 'deployed')
+
+
+def main():  # pragma: no cover
+    """
+    Main entry point.
+    """
+    import argparse
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-c', '--config', type=str,
+        help='Configuration file to use.')
+    parser.add_argument(
+        '--bus-exchange', type=str, default='commissaire',
+        help='Message bus exchange name.')
+    parser.add_argument(
+        '--bus-uri', type=str, metavar='BUS_URI',
+        default='redis://127.0.0.1:6379/',  # FIXME: Remove before release
```

Comment:
Might be nitpicking, but constants or setting values can be extracted. especially in this case it is a necessity...